src/main/java/edu/washington/iam/registry/rp/DBMetadata.java

```diff
@@ -79,8 +80,8 @@ public RelyingParty getRelyingPartyById(String id) throws RelyingPartyException
             return rps.get(0);
         }
         else {
-            String errorMsg = String.format("error getting rp: %s, rps size = %s", id, rps.size());
-            log.debug(errorMsg);
+            String errorMsg = String.format("error getting rp: %s, rps size = %s", id, (rps == null) ? 0 : rps.size());

```

Comment:
Optional nit: Extract into a local variable.

It can help to minimize computations done in format statements, so that readers can focus on understanding the format string.

Reply:
Going to ignore this for now. This error message is almost never used; I made the smallest update possible to get it to not NullPointerException.